src/main/java/com/blackducksoftware/integration/hub/dataservices/parallel/ParallelResourceProcessor.java

```diff
@@ -0,0 +1,88 @@
+package com.blackducksoftware.integration.hub.dataservices.parallel;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorCompletionService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadFactory;
+
+import com.blackducksoftware.integration.hub.dataservices.ItemTransform;
+import com.blackducksoftware.integration.log.IntLogger;
+
+public class ParallelResourceProcessor<R, S> {
+	private final Map<Class<?>, ItemTransform<List<R>, S>> transformerMap = new HashMap<>();;
+	private final ExecutorService executorService;
+	private final ExecutorCompletionService<List<R>> completionService;
+	private final IntLogger logger;
+
+	public ParallelResourceProcessor(final IntLogger logger) {
```

Comment:
You might want to allow for a client to construct their own ExecutorService stuff and pass it in through another constructor